File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.swift
class Solution {
    func maxProfit(_ prices: [Int]) -> Int {
        var max_profit = 0;
        var min_price = Int.max;

        for price in prices {
            if (price < min_price){
                min_price = price
            }
            let profit: Int = price - min_price
            max_profit = max(max_profit, profit)
        }
        return max_profit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: list[int]) -> int:
        if(len(prices) <= 1):
            return 0
        
        min_price = prices[0]
        max_profit = 0
        for price in prices[1:]:
            profit = price - min_price
            max_profit = max(max_profit, profit)
            min_price = min(min_price, price)
        return max_profit 